src/include/txn_inline.h

```diff
@@ -1352,16 +1360,39 @@ __wt_txn_modify_check(
     if (!rollback && upd == NULL) {
         __wt_read_cell_time_window(cbt, &tw, &tw_found);
         if (tw_found) {
-            if (WT_TIME_WINDOW_HAS_STOP(&tw))
+            if (WT_TIME_WINDOW_HAS_STOP(&tw)) {
                 rollback = !__wt_txn_tw_stop_visible(session, &tw);
-            else
+                if (rollback)
+                    __wt_verbose_debug(session, WT_VERB_TRANSACTION,
+                      "Conflict with update at stop timestamp: %s",
+                      __wt_timestamp_to_string(tw.stop_ts, ts_string));
+            } else {
                 rollback = !__wt_txn_tw_start_visible(session, &tw);
+                if (rollback)
+                    __wt_verbose_debug(session, WT_VERB_TRANSACTION,
+                      "Conflict with update at start timestamp: %s",
+                      __wt_timestamp_to_string(tw.start_ts, ts_string));
+            }
         }
     }
 
     if (rollback) {
         WT_STAT_CONN_DATA_INCR(session, txn_update_conflict);
         ret = __wt_txn_rollback_required(session, "conflict between concurrent operations");
+
+        /* Dump information about the txn snapshot. */
+        if (WT_VERBOSE_LEVEL_ISSET(session, WT_VERB_TRANSACTION, WT_VERBOSE_DEBUG)) {
+            if (txn->snapshot_count > 0) {
+                WT_ERR(__wt_scr_alloc(session, 1024, &buf));
+                WT_ERR(__wt_buf_fmt(session, buf, WT_SYSTEM_CKPT_SNAPSHOT "=["));
```

Comment:
The name WT_SYSTEM_CKPT_SNAPSHOT doesn't suit here. As it is not a checkpoint snapshot. You can replace it with `snapshot`? Also, can you print the `snap_min` and `snap_max`?

Reply:
I was wondering the same but then the definition of `WT_SYSTEM_CKPT_SNAPSHOT` is:
`#define WT_SYSTEM_CKPT_SNAPSHOT "snapshots"`
I think the name `WT_SYSTEM_CKPT_SNAPSHOT` is a bit misleading.

Reply:
Added them in 067f01b762789a58dbc6ecaf34a45481195f7aba